[Tay Chin Heng] iP

dozenmatter opened the pull request.

dozenmatter committed:
Added "Level-0: Greet"

dozenmatter committed:
Add Increment "Level-0: Greet"

dozenmatter committed:
Merge remote-tracking branch 'origin/master' into master

dozenmatter committed:
Add Increment "Level-1: Greet, Echo, Exit"

dozenmatter committed:
Add Increment "Level-2: Add, List"

dozenmatter committed:
Add Increment "Level-3: Mark as Done"

dozenmatter committed:
Add Increment "Level-4: ToDo, Event, Deadline"

dozenmatter committed:
Add A-TextUiTesting

dozenmatter committed:
Refactor code to "A-CodeQuality"
Add changes to Automated Text-UI Testing

thatseant commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String description;
```
Do we have to use protected here? Would using private suffice?

dozenmatter replied:
Yes I forgot to change this to private. Thanks for pointing that out!

thatseant commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
 public class Duke {
+    /* List of output messages shown to users */
+    // Greetings
+    private static final String MESSAGE_LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String MESSAGE_INTRO = "Hello from\n" + MESSAGE_LOGO;
+    private static final String MESSAGE_BODY = "Hello! I'm Duke\n"
+            + "What can I do for you?";
+    private static final String MESSAGE_GOODBYE = "Bye. Hope to see you again soon!";
+    private static final String DIVIDER = "____________________________________________________________";
+
+    // Tasks related messages
+    private static final String MESSAGE_LIST_TASK = "Here are the tasks in your list:";
+    private static final String MESSAGE_DONE_TASK = "Nice! I've marked this task as done:";
+    private static final String MESSAGE_ADD_TASK = "Got it. I've added this task:";
+
+    /* List of user input commands */
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_BYE = "bye";
+
+    /* List of user command parameters */
+    private static final String PARAM_AT = "/at";
+    private static final String PARAM_BY = "/by";
```
I like how all command keywords are defined as constants at the start of the file so they can be easily changed when necessary.

thatseant commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcome();
+        while (SC.hasNextLine()) {
+            String input = getUserInput();
+            executeUserCommand(input);
+        }
```
I like how your code is abstracted into functions, enhancing readability without need for comments. Good Job! 👍

thatseant reviewed (commented):
Good Job! Your code readability is good and follows coding guidelines!

Colin386 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+        return by;
+    }
+
+    public void setBy(String by) {
```
LGTM

Colin386 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+public class Deadline extends Task {
```
LGTM

Colin386 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String MESSAGE_INTRO = "Hello from\n" + MESSAGE_LOGO;
+    private static final String MESSAGE_BODY = "Hello! I'm Duke\n"
+            + "What can I do for you?";
```
May I suggest to put the line 13 and 14 into a single line or perhaps to ensure the intentation is around 4 spaces

dozenmatter replied:
Makes sense. Thanks for your suggestion!

Colin386 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+        String[] processedInputs = processUserInput(input);
+        String command = processedInputs[0];
+        String args = processedInputs[1];
+
```
Looks good! I really like the use of constants in switch statements. It makes things clear

Colin386 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+            exitProgram();
+        }
+    }
+
```
Perhaps for comments of function features, you can add a blank line to separate the function descriptions from the parameters

Colin386 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,228 @@
+        TASKS.add(task);
+        printAddTask(task);
+    }
+
```
I like how you verify the task number before running

Colin386 reviewed (commented):
Overall it looks good! No coding violations seem to have been detected

dozenmatter committed:
Add Increment "Level 5: Handle Errors"
Add Extension "A-Exceptions"

dozenmatter committed:
Merge branch 'branch-Level-5' into master

dozenmatter committed:
Modify code to handle DukeException

dozenmatter committed:
Merge branch 'branch-Level-5' into master

dozenmatter committed:
Modify processTaskAsDone() to throw DukeException
Modify executeUserCommand() to handle DukeException

dozenmatter committed:
Merge branch 'branch-Level-5' into master

dozenmatter committed:
Add new package "duke"
Move all classes into "duke" package
Move task related classes into duke.task package
Modify runtest.bat accordingly

dozenmatter committed:
Merge branch 'branch-A-Packages' into master

dozenmatter committed:
Add Increment "Level-6: Delete"
Add "A-Collections"
Add delete function for tasks
Modify input.txt with updated inputs
Modify EXPECTED.TXT outputs

dozenmatter committed:
Add Increment "Level-7: Save"
Add a FileManager class
Modify runtest.bat file

dozenmatter committed:
Merge branch 'branch-Level-6' into master

dozenmatter committed:
Merge branch 'branch-Level-7' into master

# Conflicts:
#	src/main/java/duke/Duke.java

dozenmatter committed:
Refactor codes
Refactor Task class to abstract class
Rename FileManager class as "Storage"
Refactor Storage method names into load() and save()
Rename package duke.file to duke.store
Modify runtest.bat

dozenmatter committed:
Refactor codes in Storage.java

dozenmatter committed:
Refactor codes to "A-MoreOOP"

dozenmatter committed:
Add Increment "Level 9: Find"
Add a new FindCommand class

dozenmatter committed:
Add Increment "Level 8: Dates and Times"
Add a new DateTimeParser interface

dozenmatter committed:
Merge pull request #1 from dozenmatter/branch-Level-8

Add Increment "Level 8: Dates and Times"

dozenmatter committed:
Merge branch 'master' into branch-Level-9

dozenmatter committed:
Remove unused import in FindCommand.java

dozenmatter committed:
Merge pull request #2 from dozenmatter/branch-Level-9

Branch level 9

dozenmatter committed:
Update README.md

Update Usage list

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Add an additional DIVIDER to printWelcome()

dozenmatter committed:
Merge remote-tracking branch 'origin/master' into master

dozenmatter committed:
Update README.md

jtankw3 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,141 @@
+
+    /**
+     * Displays a message enclosed within dividers.
+     * @param message user specified message(s)
```
There should be an empty line between the description and parameter section of your javadoc

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,78 @@
+     *
+     * @param args user specified arguments
+     */
+    public static void main(String[] args) {
```
You may want to put the main method closer to the top as it is what readers are looking out for.

jtankw3 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    /**
+     * Takes in task list size and checks if user specified task number is valid
```
Do not have to write what it takes in to make the comment more concise.

jtankw3 commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,58 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (isValidTaskNumber(tasks.getSize())) {
+            Task t = tasks.get(taskNum);
```
Although `t` has a small scope, we can try to give it a better name.

jtankw3 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.printSearchResult(tasks, keyword);
```
Why do you make `ui` do the finding instead of having the logic here?

jtankw3 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,68 @@
+     * @return the command based on user input
+     * @throws DukeException if user command is invalid
+     */
+    public static Command parse(String input) throws DukeException {
```
Method name can be clearer by indicating what it parses, as the parser could potential parse many different things.

jtankw3 commented on src/main/java/duke/store/Storage.java:
```diff
@@ -0,0 +1,74 @@
+ * Saving and loading of saved data
+ */
+public class Storage {
+    private static final String CURRENT = System.getProperty("user.dir");
```
To be clearer, variable names should be nouns

jtankw3 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,144 @@
+     * @param line string of tasks
+     * @throws DukeException if string cannot be parsed
+     */
+    public void parseToArrayList(String line) throws DukeException {
```
Generally we do not want to expose that what data structure we are using in the method name. This is in case we want to change the implementation in future to another data structure, we cannot do that as all code using this method name will become obsolete.

jtankw3 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,144 @@
+
+            tasks.add(t);
+        } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+            throw new DukeException(ERROR_READ_DATA);
```
May not be the best error message as this method can be used for other purposes besides reading files.

jtankw3 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,52 @@
+import java.util.Date;
+
+/**
+ * Handles Event tasks.
```
The class does not really "Handle" the task.

jtankw3 reviewed (commented):
Really great job, the division into different classes is well done and there are very few coding standard issues, making everything very organized and readable. Keep up the good work 👍

dozenmatter committed:
Refactor code to adhere coding standards and better readability.

dozenmatter committed:
Merge remote-tracking branch 'origin/master' into master

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Add Increment "A-UserGuide"

dozenmatter committed:
Merge remote-tracking branch 'origin/master' into master

dozenmatter committed:
Set theme jekyll-theme-cayman

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Update README.md

dozenmatter committed:
Add Increment "A-UserGuide"

dozenmatter committed:
Merge remote-tracking branch 'origin/master' into master

# Conflicts:
#	docs/README.md

dozenmatter committed:
Fix a bug in "find" command

dozenmatter committed:
Update README.md
